astropy 1.0.5, failed test in test_sky_coord.py, only in i386

I'm seeing the following test fail with Python 2.7, numpy 1.10.0b1 and i386 architecture. In x86_64 this same test passes

```
platform linux2 -- Python 2.7.10 -- py-1.4.30 -- pytest-2.7.3
Running tests with Astropy version 1.0.5.
Running tests in astropy.
Platform: Linux-4.1.7-200.fc22.x86_64-i686-with-fedora-24-Rawhide
Executable: /usr/bin/python2
Full Python Version: 
2.7.10 (default, Sep 23 2015, 07:01:59) 
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
encodings: sys: ascii, locale: ANSI_X3.4-1968, filesystem: ANSI_X3.4-1968, unicode bits: 20
byteorder: little
float info: dig: 15, mant_dig: 15
Numpy: 1.10.0b1
Scipy: 0.14.1
Matplotlib: 1.4.3
h5py: 2.5.0

    @pytest.mark.skipif('NUMPY_LT_1_7')
    def test_repr_altaz():
        sc2 = SkyCoord(1 * u.deg, 1 * u.deg, frame='icrs', distance=1 * u.kpc)
        loc = EarthLocation.from_geodetic(-122*u.deg, -47*u.deg)
        time = Time('2005-03-21 00:00:00')
        sc4 = sc2.transform_to(AltAz(location=loc, obstime=time))
>       assert repr(sc4).startswith("<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, "
                             "location=(-2309222.664660742, -3695528.7655007695, "
                             "-4641764.788820372) m, pressure=0.0 hPa, "
                             "temperature=0.0 deg_C, relative_humidity=0, "
                             "obswl=1.0 micron): (az, alt, distance) in "
                             "(deg, deg, m)\n")
E       assert <built-in method startswith of str object at 0xefdf9520>('<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, location=(-2309222.664660742, -3695528.7655007695, -4641764.788820...pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)\n')
E        +  where <built-in method startswith of str object at 0xefdf9520> = '<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, location=(-2309222.6646607416, -3695528.765500769, -4641764.788820...y=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)\n    (297.31930986, 21.87980592, 30856775790428696576.0)>'.startswith
E        +    where '<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, location=(-2309222.6646607416, -3695528.765500769, -4641764.788820...y=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)\n    (297.31930986, 21.87980592, 30856775790428696576.0)>' = repr(<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, location=(-2309222.66466074...ance) in (deg, deg, m)\n    (297.31930986, 21.87980592, 30856775790428696576.0)>)

astropy/coordinates/tests/test_sky_coord.py:508: AssertionError
```

--- a/astropy/coordinates/tests/test_sky_coord.py
+++ b/astropy/coordinates/tests/test_sky_coord.py
@@ -503,12 +503,12 @@ def test_repr():
 @pytest.mark.skipif('NUMPY_LT_1_7')
 def test_repr_altaz():
     sc2 = SkyCoord(1 * u.deg, 1 * u.deg, frame='icrs', distance=1 * u.kpc)
-    loc = EarthLocation.from_geodetic(-122*u.deg, -47*u.deg)
+    loc = EarthLocation(-2309223 * u.m, -3695529 * u.m, -4641767 * u.m)
     time = Time('2005-03-21 00:00:00')
     sc4 = sc2.transform_to(AltAz(location=loc, obstime=time))
     assert repr(sc4).startswith("<SkyCoord (AltAz: obstime=2005-03-21 00:00:00.000, "
-                         "location=(-2309222.664660742, -3695528.7655007695, "
-                         "-4641764.788820372) m, pressure=0.0 hPa, "
+                         "location=(-2309223.0, -3695529.0, "
+                                "-4641767.0) m, pressure=0.0 hPa, "
                          "temperature=0.0 deg_C, relative_humidity=0, "
                          "obswl=1.0 micron): (az, alt, distance) in "
                          "(deg, deg, m)\n")

